oidc-authenticator/src/main/java/org/xwiki/contrib/oidc/auth/internal/OIDCUserManager.java

```diff
@@ -143,6 +163,79 @@ public class OIDCUserManager
 
     private static final String XWIKI_GROUP_PREFIX = "XWiki.";
 
+    private ConfigurableJWTProcessor<SecurityContext> jwtProcessor;
+
+    @Override
+    public void initialize() throws InitializationException
+    {
+        jwtProcessor = new DefaultJWTProcessor<>();
+
+        try {
+            jwtProcessor.setJWSTypeVerifier(new DefaultJOSEObjectTypeVerifier<>(new JOSEObjectType("JWT")));
+            JWKSource<SecurityContext> keySource;
+            keySource = JWKSourceBuilder.create(configuration.getClientProvider().getMetadata().getJWKSetURI().toURL())
+                .retrying(true).build();
+
+            JWSAlgorithm expectedJWSAlg = JWSAlgorithm.RS256;
+            // Configure the JWT processor with a key selector to feed matching public
+            // RSA keys sourced from the JWK set URL
+            JWSKeySelector<SecurityContext> keySelector = new JWSVerificationKeySelector<>(expectedJWSAlg, keySource);
+            jwtProcessor.setJWSKeySelector(keySelector);
+        } catch (IOException | GeneralException | URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public XWikiUser checkAccessToken(String idTokenHeader, String accessTokenHeader)
+    {
+        try {
+            DocumentReference ref = userByTokenCache.get(Pair.of(idTokenHeader, accessTokenHeader));
```

Comment:
Not sure why you need the id token for the cache key, the access token is already supposed to be specific to a user.